### ISSUE TITLE
[Bug]

### ISSUE THREAD

**Issue by goodluckhavefun123** [tagpro-game/tagpro-issues#112]
I will randomly get a DC for some reason. Please Fix

**Comment by steppin** [comment]
Thanks for the report.  If you can, it makes it easier for us to view bugs if title gives a short summary of the issue.  For this one for example, something like "randomly kicked out of a game" would be great.

**Comment by steppin** [comment]
thanks :+1:

**Comment by NugaSausage** [comment]
I'm also getting randomly kicked too. Sometimes the Ping and Loss meters will suggest lag, but also it happens randomly and suddenly too.

This especially sucks when compounded with the 10 second wait for "refreshing into a game after the first 15 seconds" clause.

**Comment by NewCompte** [comment]
Did you use to have >5 second lag sometimes ?

We put code recently which makes the server disconnect you when you don't communicate for 5-7 seconds.

**Comment by NugaSausage** [comment]
Yeah, it was short.

I thought the AFK timer was reduced from a minute to 30 seconds. Just asking.

**Comment by goodluckhavefun123** [comment]
No, I just got randomly kicked in the middle of games with no lag.

**Comment by goodluckhavefun123** [comment]
This just happened again WITH NO LAG BEFORE. Please fix.

**Comment by goodluckhavefun123** [comment]
AND AGAIN

**Comment by goodluckhavefun123** [comment]
AND AGAIN. This is really getting out of hand (five times today). Starting to have a serious effect on make rank points (going for weekly)

**Comment by steppin** [comment]
this is hard to troubleshoot because we cannot reproduce it.  Can you play with developer tools open, select preserve log in console, and see if anything interesting shows up when you are getting kicked?

**Comment by goodluckhavefun123** [comment]
what is developer tools/where do I open it

**Comment by SomeBall-1** [comment]
@goodluckhavefun123 ctrl+shft+j for windows and cmd+opt+j for macs and I think internet explorer is F12 if you're using that. A little window will either open as part of your current tab or as a separate window, if you see [this picture](http://tagpro-radius.koalabeast.com/images/console.png) then you did it right.

@NugaSausage He means not communicating with the server as in the server is receiving no information from the client, it's independent of user input.

**Comment by banzaiTP** [comment]
Hey folks, I'm asking this in here because while I don't think my issue is quite the same, it does lead to me getting kicked. If this violates any rules or is out of place, obviously feel free to delete it. 

Once in a while, I'll just completely freeze up for 5-15 seconds. I don't "lag out." The whole browser essentially freezes. Or rather it's unresponsive. I can move my mouse, but I can't control the ball/chat/click exit/etc. It used to not be that big of a deal, because once it comes back I'm fine. But lately, if it lasts long enough, obviously I DC.

I've always assumed it's on my end, but I have no idea why it happens. Only extensions I have now are a texture pack/ball spin. I'm clueless about this stuff; I don't know if it's a connection issue, if my computer can't handle the prolonged workload of several games in a row, or what. I don't play any other games, so I don't know if it would happen with other similarly "intensive" things happening on my browser.

Anyway, I'm not asking you to fix anything for me, I'm just curious if you could give me a general idea of what it is that's "going wrong." (Googling things like "browser freezing" obviously brings up so many different results that it's hard for me to find an issue specifically related to what I experience.) I really appreciate it. Thanks!

**Comment by dddeeefff** [comment]
Hi.

I have this issue. It's very frustrating and is causing a lot of DCs.

Chrome: Version 43.0.2357.81
User scripts: none
Extensions: adBlock 
```
Navigated to http://tagpro-chord.koalabeast.com:8000/
tagpro-chord.koalabeast.com/:20 GET http://pagead2.googlesyndication.com/pagead/show_ads.js net::ERR_BLOCKED_BY_CLIENT
(index):9 GET http://www.google-analytics.com/analytics.js net::ERR_BLOCKED_BY_CLIENT(anonymous function) @ (index):9(anonymous function) @ (index):10
/compact/global-game.js:1 TagPro version 3.2.1
global-game.js:1 Powered by PixiJS. Currently running in WebGL mode
global-game.js:1 Developers: LuckySpammer, steppin, NewCompte, AnkhMorpork, Ly, ylambda
global-game.js:1 +
8global-game.js:1 PixiParticles: using particle textures from different images may hinder performance in WebGL
Navigated to http://tagpro-chord.koalabeast.com/?err=ping%20timeout
```

I'm pretty confident that ERR_BLOCKED_BY_CLIENT is adblock, so let's ignore those; they don't seem to be affecting the game anyway.

The "PixiParticles" warning appears in every session. Doesn't appear to have any impact.

I don't think there's anything in the logs that would help solve this issue. How does the redirect work? Any chance you could do a variable dump in the console before it does it?
